src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -474,4 +524,16 @@ private function handleChangeOrderStatusException(ChangeOrderStatusException $e)
             );
         }
     }
+
+    /**
+     * @param Request $request
+     * @return int|null
+     */
+    private function addCustomizationFields(Request $request): ?int
+    {
+        $customizationFields = $request->request->get(self::PRODUCT_CUSTOMIZATION_FIELDS_REQUEST_KEY);
```

Comment:
Hey @matks i need your guidance :) For add product to cart component in BO order creation i have to add product customization fields via ajax. They can be added by user in product page and can be file or text type. Im thinking of creating a separate service for validating and resolving data of these fields and use command to actually add them to database with legacy functions. Other idea would be to use form with collection type fields. Although form imo would look cleaner, i am not sure if it is going to work. Maybe you have some suggestions or insights about that?

Reply:
I think I answered that 🤔 maybe in gitter ?